src/overlay/TxDemandsManager.cpp

```diff
@@ -269,6 +274,27 @@ TxDemandsManager::recvTxDemand(FloodDemand const& dmd, Peer::pointer peer)
     ZoneScoped;
     auto& herder = mApp.getHerder();
     auto& om = mApp.getOverlayManager().getOverlayMetrics();
+#ifdef BUILD_TESTS
+    auto msg = std::make_shared<StellarMessage>();
+    size_t batchSize = 0;
+    if (mApp.getConfig().EXPERIMENTAL_TX_BATCH_MAX_SIZE > 0)
+    {
+        msg->type(TX_SET);
+        msg->txSet().previousLedgerHash = OverlayManager::txBatchHash();
+    }
+
+    auto sendAndReset = [&](std::shared_ptr<StellarMessage> msg) {
+        if (batchSize > 0)
+        {
+            om.mTxBatchSizeHistogram.Update(batchSize);
+            peer->sendMessage(std::move(msg));
+            msg = std::make_shared<StellarMessage>();
+            msg->type(TX_SET);
+            msg->txSet().previousLedgerHash = OverlayManager::txBatchHash();
```

Comment:
Might be nice to refactor this into a small helper function like "createTxBatchMessage", with a comment explaining the structure just so we have all the weirdness in one documented place.

Reply:
Good idea, done.